cmd/kogito/command/shared/converters.go

```diff
@@ -19,14 +19,32 @@ import (
 	"github.com/kiegroup/kogito-cloud-operator/pkg/util"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
+	"strings"
 )
 
+const secretNameKeySeparator = "#"
+
 // FromStringArrayToEnvs converts a string array in the format of key=value pairs to the required type for the Kubernetes EnvVar type
-func FromStringArrayToEnvs(strings []string) []v1.EnvVar {
-	if strings == nil {
+func FromStringArrayToEnvs(keyPairStrings []string) []v1.EnvVar {
+	if len(keyPairStrings) == 0 {
 		return nil
 	}
-	return framework.MapToEnvVar(util.FromStringsKeyPairToMap(strings))
+	var envVars []v1.EnvVar
+	keyPairMap := util.FromStringsKeyPairToMap(keyPairStrings)
+	for key, value := range keyPairMap {
+		if strings.Contains(value, secretNameKeySeparator) {
```

Comment:
one question: what happens if the value of an env var contains a `#` and is not supposed to be a secret, like a password for example ?
the value could contain a `#` where as the key should not as it should follow the env var naming convention.

Shouldn't we instead use a specific "equality" like for example `#=` (that can be something else) so we are sure that it is a secret name/key that is set for the variable ?
that would give us `MY_VAR#=secretName#secretKey`

Reply:
Hmm, wouldn't it rather have sense to create a separate flag for variable from secret?
That way we can keep a flexibility in case we will need to add another variable source, for example config map.

Reply:
@vaibhavjainwiz have you discussed this feature with @MarianMacik? Since he asked for this, I believe he could weight in. @radtriste has a good point tho.

Reply:
> Hmm, wouldn't it rather have sense to create a separate flag for variable from secret?
> That way we can keep a flexibility in case we will need to add another variable source, for example config map.

@sutaakar I came to thought to introduce seperate variable to capture secrets but after analysis of code I found that we already have two different flag to capture envs(--env & --build-env) and user can provide secret in either of them. So if I go with this approch then we have add two more flag(`--env-secret & --build-env-secret`) which looks like too many flags just to capture env variable.

Reply:
> @vaibhavjainwiz have you discussed this feature with @MarianMacik? Since he asked for this, I believe he could weight in. @radtriste has a good point tho.

@ricardozanini Yes, I represent my approach to @MarianMacik on Jira and took his confirmation.

Reply:
@sutaakar @ricardozanini @radtriste Please suggest How we should processed further with this. Should I created new seperate flag to capture secret, chage secret seperator or leave it as it is?

Reply:
@vaibhavjainwiz I'd say to let it as is and open a new JIRA to dive into the discussion, but since @radtriste brought this up, I'll give him the final word.

Reply:
Honestly, I would not let it like that as this can be problematic.
If we need a solution for now, I would go for @sutaakar idea, with many options. This can be refactored later and is more visible to not forget.
It makes more options for sure but at least, we know exactly what it does and there is no specific convention to write a secret in the `--env` option.

Reply:
ok then, I go with @sutaakar idea to introduce new variables

Reply:
done